### PR TITLE
PA-1484: Change `IsS3BucketEmpty()` to `IsBackupLocationEmpty()` to support nfs 

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
- [x] Change `IsS3BucketEmpty()` to `IsBackupLocationEmpty()` to support nfs

**Which issue(s) this PR fixes** (optional)
Closes # PA-1484

 Dashboard URL : https://aetos.pwx.purestorage.com/resultSet/testSetID/297552 

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_restore_basic_test.go [hunk 1]
```diff
@@ -2016,7 +2016,7 @@ var _ = Describe("{CloudSnapsSafeWhenBackupLocationDeleteTest}", func() {
 				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
 				customBucket = GetCustomBucketName(provider, "cloudsnapssafewhenbackuplocationdelete")
 				// Check if custom bucket is empty or not
-				result, err := IsS3BucketEmpty(customBucket)
+				result, err := IsBackupLocationEmpty(provider, customBucket)
 				log.FailOnError(err, "Checking for errors while checking s3 bucket")
```
Comment (vpinisetti-px) [inline tests/backup/backup_restore_basic_test.go:2020]: Better to change log also instead of hardcoding to S3

**Review by vpinisetti-px — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

**Comment by kshithijiyer-px** [comment]
URL https://aetos.pwx.purestorage.com/resultSet/testSetID/297347

**Comment by mkoppal-px** [comment]
Both the runs you have attached with s3 and not NFS

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**